common/ethereum/src/eth_contracts/ptokens_router.rs

```diff
@@ -0,0 +1,108 @@
+use std::convert::TryFrom;
+
+use common::{types::Bytes, AppError};
+use common_metadata::{MetadataChainId, METADATA_CHAIN_ID_NUMBER_OF_BYTES};
+use derive_more::Constructor;
+use ethabi::{decode as eth_abi_decode, ParamType as EthAbiParamType, Token as EthAbiToken};
+
+use crate::{EthLog, EthLogExt, EthReceipt};
+
+crate::make_topics!(
+    PTOKENS_ROUTER_METADATA_EVENT_TOPIC => "41954c3bf6e497b17fc12f429900878df830619bbcccb5f61aedc91e6ccc9e64",
+);
+
+#[derive(Debug, Clone, Constructor, Eq, PartialEq, Default)]
+pub struct PTokensRouterMetadataEvent {
+    pub user_data: Bytes,
+    pub origin_chain_id: MetadataChainId,
+    pub origin_address: String,
+    pub destination_chain_id: MetadataChainId,
+    pub destination_address: String,
```

Comment:
This isn't wrong per se, it's just the older way I was doing this. Now I prefer to have more control by keeping struct members private and providing getters instead. 

As such, the crate `derive_getters` (it's in the workspace so can be imported to this crate via that) makes it v simple. 

So these tagged lines would then all lose their `pub`. 

This way we thin out the API and allow ourselves to change the internals of the struct in future, potentially _without_ having breaking changes.